### PR TITLE
Feat file search

### PR DESCRIPTION
**Pull request by Karim94**
[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author
Karim
#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7519
#### Linked PRs in other repos
https://github.com/PeerioTechnologies/peerio-icebear/pull/77
#### Manual testing instructions  
N/A
#### Additional notes
N/A

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner

- [ ] At least one approved code review exists.
- [ ] Approved tester review exists and was done after last code changes.
- [ ] CI build and tests passed.
- [ ] This feature/bugfix is considered fully done and can get merged.
- [ ] PR name follows conventional changelog format.
- [ ] I'll remember to squash commits.

### CODE REVIEW

File: app/components/files/files.js [hunk 1]
```diff
@@ -124,14 +130,104 @@ export default class Files extends SafeComponent {
...
+    searchFile = val => {
+        if (val === '' || val === null) {
+            fileState.store.clearFilter();
+            return;
+        }
+        fileState.store.filterByName(val);
+    };
+
+    searchTextbox() {
+        const height = 32;
```
Comment (seavan) [inline app/components/files/files.js]: vars

File: app/components/files/files.js [hunk 2]
```diff
@@ -124,14 +130,104 @@ export default class Files extends SafeComponent {
...
+            marginHorizontal: vars.spacing.medium.mini2x,
+            marginVertical: vars.spacing.small.midi,
+            borderColor: vars.verySubtleGrey,
+            borderWidth: 1,
+            height,
+            borderRadius: height
+        };
+        const placeholderStyle = {
+            flexGrow: 1,
+            height: 48,
```
Comment (seavan) [inline app/components/files/files.js]: vars

File: app/components/files/files.js [hunk 3]
```diff
@@ -124,14 +130,104 @@ export default class Files extends SafeComponent {
...
-        const body = (this.data.length || !fileState.currentFolder.isRoot) ?
-            this.listView() : !fileState.store.loading && <FilesPlaceholder />;
+        let body;
+        if (this.data.length || !fileState.currentFolder.isRoot) body = this.listView();
+        else if (!this.data.length && this.findFilesText && !fileState.store.loading) {
+            body = (
+                <Text style={{ marginTop: vars.headerSpacing, textAlign: 'center' }}>
+                    {tx('title_noFilesMatchSearch')}
+                </Text>);
+        } else body = !fileState.store.loading && <FilesPlaceholder />;
```
Comment (seavan) [inline app/components/files/files.js]: can it be decomposed so it doesn't have else if and two elses and overall looks more clear?

Comment (Karim94) [reply app/components/files/files.js]: Pushed, not sure if its that much clearer now though.

File: app/components/files/files.js [hunk 4]
```diff
@@ -124,18 +130,113 @@ export default class Files extends SafeComponent {
...
+        if (!this.data.length && this.findFilesText && !fileState.store.loading) {
+            return (
+                <Text style={{ marginTop: vars.headerSpacing, textAlign: 'center' }}>
+                    {tx('title_noFilesMatchSearch')}
+                </Text>
+            );
+        }
+        return (
+            !fileState.store.loading && <FilesPlaceholder />
+        );
```
Comment (seavan) [inline app/components/files/files.js]: braces are not needed here, can be all put in one line

Comment (Karim94) [reply app/components/files/files.js]: Done

**Comment by Karim94** [comment]
Tested by Eren and ready for merge @seavan

File: package.json [hunk 5]
```diff
@@ -47,7 +47,7 @@
     "path": "0.12.7",
     "pbkdf2": "github:PeerioTechnologies/pbkdf2",
     "peerio-copy": "github:PeerioTechnologies/peerio-copy#icebear",
-    "peerio-icebear": "github:PeerioTechnologies/peerio-icebear#fix-hotfix-platformchange-error",
+    "peerio-icebear": "github:PeerioTechnologies/peerio-icebear#feat-file-search",
```
Comment (seavan) [inline package.json]: if it's ready to merge, this should point to icebear version

**Comment by seavan** [comment]
@ebilg waiting for your testing to be merged.

**Pull request merged**